chain/client/src/client_actor.rs

```diff
@@ -807,24 +807,20 @@ impl Handler<Status> for ClientActor {
                     .chain
                     .blocks_with_missing_chunks
                     .list_blocks_by_height(),
-                epoch_info: EpochInfoView {
-                    epoch_id: head.epoch_id.0,
-                    height: epoch_start_height,
-                    first_block_hash: self
-                        .client
-                        .chain
-                        .get_block_by_height(epoch_start_height)?
-                        .header()
-                        .hash()
-                        .clone(),
-                    start_time: self
-                        .client
-                        .chain
-                        .get_block_by_height(epoch_start_height)?
-                        .header()
-                        .timestamp()
-                        .to_rfc3339(),
-                    validators: validators.to_vec(),
+                epoch_info: {
+                    let maybe_epoch_start_block =
+                        self.client.chain.get_block_by_height(epoch_start_height);
+                    EpochInfoView {
+                        epoch_id: head.epoch_id.0,
+                        height: epoch_start_height,
+                        first_block: match maybe_epoch_start_block {
+                            Ok(block) => {
+                                Some((block.header().hash().clone(), block.header().timestamp()))
+                            }
+                            Err(_) => None,
+                        },
```

Comment:
Since you’re going with an `Option<(..., ...)>` this can now be simplified to:

```rust
self.client
    .chain
    .get_block_by_height(epoch_start_height)
    .ok()
    .map(|block| (block.header().hash().clone(), block.header().timestamp()));
```